include/pacemaker.h

```diff
@@ -186,6 +186,41 @@ int pcmk_list_nodes(xmlNodePtr *xml, char *node_types);
  */
 int pcmk_status(xmlNodePtr *xml);
 
+/*!
+ * \brief Display the name and/or description of a result code
+ *
+ * \param[in,out] xml        The destination for the result, as an XML tree
+ * \param[in]     code       The result code
+ * \param[in]     type       Interpret \c code as this type of result code
+ *                           Supported values: \c pcmk_result_legacy,
+ *                           \c pcmk_result_pcmk_rc,
+ *                           \c pcmk_result_crm_exitcode.
+ * \param[in]     show_name  Whether to show the result code's name
+ * \param[in]     show_desc  Whether to show the result code's description
+ *
+ * \return Standard Pacemaker return code
+ *
+ * \note At least one of \c show_name and \c show_desc must be set to \c true.
+ */
+int pcmk_show_result_code(xmlNodePtr *xml, int code, pcmk_result_type_t type,
+                          bool show_name, bool show_desc);
+
+/*!
+ * \brief List all valid result codes in a particular family
+ *
+ * \param[in,out] xml        The destination for the result, as an XML tree
+ * \param[in]     type       The family of result codes to list. Supported
+ *                           values: \c pcmk_result_legacy,
+ *                           \c pcmk_result_pcmk_rc,
+ *                           \c pcmk_result_crm_exitcode.
+ * \param[in]     show_name  Whether to show each result code's name
+ * \param[in]     show_desc  Whether to show each result code's description
+ *
+ * \return Standard Pacemaker return code
+ */
+int pcmk_list_result_codes(xmlNodePtr *xml, pcmk_result_type_t type,
+                           bool show_name, bool show_desc);
+
 #ifdef BUILD_PUBLIC_LIBPACEMAKER
```

Comment:
I'm going to leave it up to @kgaillot to decide whether these things should be above or below BUILD_PUBLIC_LIBPACEMAKER.

I find it a little confusing that we have both `pcmk_` functions and that macro guard, when anything that's `pcmk_` is already public API.  That fact means these new functions would still be public, which means changing to a set of flags later would be difficult.  So it's probably worth starting with flags.

Reply:
Ah fair enough. I turned them into flags; they'll go in the next update.

Agreed about `BUILD_PUBLIC_LIBPACEMAKER`. I think commit 27677a6d drew it away from its intended purpose.

Reply:
The stonith_admin API was the first one to be added to libpacemaker, and it wasn't ready for release (still isn't), so BUILD_PUBLIC_LIBPACEMAKER was added to prevent it from going public just yet, even though it is intended as public API eventually.

Then we added other APIs that were final, so they didn't need to be guarded, and the name became misleading. Something like PCMK__ENABLE_FENCE_API would be better now.

FYI the main thing holding up the fencing API is that we don't want to drag stonith_t into libpacemaker. Eventually the fencer will be converted to use pcmk_ipc_api_t instead, and then we can decide what the public API should look like. Another possibility would be to just drop the stonith_t arguments and have each public API create its own connection.